### DISCUSSION TITLE
How to get started with DockerFiles?

### DISCUSSION THREAD

**Discussion by manuGil** [TU-Delft-DCC/TU-Delft-DCC.github.io#25] [Archive]
I am looking for way to gets started with DockerFiles ASAP.  Where to you recommend to start? The Docker documentation?

**Comment by manuGil** [comment]
Suggestion:
Watch this 10 mins video on the basics of what a DockerFile is and how it works: https://www.youtube.com/watch?v=LQjaJINkQXY

For reference on possible commands, check these:
- DockerFile reference from Docker: https://docs.docker.com/engine/reference/builder/ 
- A Cheatsheet: https://github.com/wsargent/docker-cheat-sheet#dockerfile

**Reply by jurra** [reply]
Hi @manuGil I believe this one can also be ver helpful: [Best practices for writing docker file](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/), also from the docker documentation website.

The jupyter stacks also share some best practices of building docker images, you can see how they build on different images as a stack and separate the images in different layers: [https://github.com/jupyter/docker-stacks](https://github.com/jupyter/docker-stacks). Each folder has a different Dockerfile that builds upon other images.
